app/tests/spec/models/user.js

```diff
@@ -674,62 +674,80 @@ define(function (require, exports, module) {
         });
       });
 
-      describe('with an unverified account', function () {
-        beforeEach(function () {
-          account = user.initAccount({ email: 'email', uid: 'uid' });
+      // Ensure that content-server can correctly handle any emailSent response
+      var emailSentTestCases = [{
+        emailSent: undefined,
+        expectedRetryCallCount: 1
+      }, {
+        emailSent: false,
+        expectedRetryCallCount: 1
+      }, {
+        emailSent: true,
+        expectedRetryCallCount: 0
+      }];
+      emailSentTestCases.forEach(function (testCase) {
+        describe('with an unverified account, emailSent = ' + testCase.emailSent, function () {
+          beforeEach(function () {
+            account = user.initAccount({email: 'email', uid: 'uid'});
 
-          sinon.stub(account, 'signIn', function () {
-            return p();
-          });
+            sinon.stub(account, 'signIn', function () {
+              return p();
+            });
 
-          sinon.stub(account, 'get', function (property) {
-            if (property === 'verified') {
-              return false;
-            } else if (property === 'verificationReason') {
-              return VerificationReasons.SIGN_UP;
-            }
+            sinon.stub(account, 'get', function (property) {
 
-            return property;
-          });
+              if (property === 'verified') {
+                return false;
+              } else if (property === 'verificationReason') {
+                return VerificationReasons.SIGN_UP;
+              } else if (property === 'emailSent') {
+                return testCase.emailSent;
+              }
 
-          sinon.stub(account, 'retrySignUp', function () {
-            return p();
-          });
+              return property;
+            });
 
-          sinon.stub(user, 'setSignedInAccount', function () {
-            return p(account);
-          });
+            sinon.stub(account, 'retrySignUp', function () {
+              return p();
+            });
 
-          sinon.spy(notifier, 'triggerRemote');
+            sinon.stub(user, 'setSignedInAccount', function () {
```

Comment:
Is it possible to use a spy instead of a stub here? `setSignedInAccount` doesn't seem like it has any side effects that should cause problems elsewhere.